### PR TITLE
[niconico] Fix downloading videos with null owner

### LINKED ISSUE #14135: New Nicovideo Issue
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.09.02*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [ ] I've **verified** and **I assure** that I'm running youtube-dl **2017.09.02**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

 youtube-dl  http://www.nicovideo.jp/watch/sm18238488 --username USER --password USER -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['http://www.nicovideo.jp/watch/sm18238488', '--username', 'PRIVATE', '--password', 'PRIVATE', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.09.02
[debug] Python version 3.5.2 - Linux-4.4.0-81-generic-i686-with-LinuxMint-18.1-serena
[debug] exe versions: ffmpeg 2.8.11-0ubuntu0.16.04.1, ffprobe 2.8.11-0ubuntu0.16.04.1
[debug] Proxy map: {}
[niconico] Logging in
[niconico] sm18238488: Downloading webpage
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/__init__.py", line 465, in main
    _real_main(argv)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/__init__.py", line 455, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/YoutubeDL.py", line 1958, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/YoutubeDL.py", line 776, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/extractor/common.py", line 434, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.5/dist-packages/youtube_dl/extractor/niconico.py", line 399, in _real_extract
    uploader_id = get_video_info(['ch_id', 'user_id']) or owner.get('id')
AttributeError: 'NoneType' object has no attribute 'get'

---

### Description of your *issue*, suggested solution and other information

I try and download a video with NicoVideo, and it gives this error. I tried to check for a recent bug report on this but i didn't see anything. Does anyone know the cause? thank you.

### PR DESCRIPTION
**Pull request by pawitp**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Niconico may return JSON where owner is set to null. In that case, api_data.get will return None even though a default is provided. Fix the check so that we always change None into an empty dict.

To run the tests, comment out the "skip" and put your Niconico account into in `test/parameters.json`.

Fixes #14135

### CODE REVIEW

File: youtube_dl/extractor/niconico.py [hunk 1]
```diff
@@ -395,7 +414,9 @@ def get_video_info(items):
 
         webpage_url = get_video_info('watch_url') or url
 
-        owner = api_data.get('owner', {})
+        # Note: cannot use api_data.get('owner', {}) because owner may be set to "null"
+        # in the JSON, which will cause None to be returned instead of {}.
+        owner = api_data.get('owner') or {}
```
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: This won't help `owner` is set to `True`-ish non dict.

Comment (pawitp) [reply youtube_dl/extractor/niconico.py]: I don't see why that would practically happen though. I imagine that they will only return two things:
- an object if a owner is present
- null if an owner is not present

This is the pattern used in other extractors too, for example https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/instagram.py#L274.

What's your suggestion?

Comment (dstftw) [reply youtube_dl/extractor/niconico.py]: You can't make any assumptions on data returned. This pattern is flawed. Suggestion is to make it work in all possible cases.

File: youtube_dl/extractor/niconico.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ class NiconicoIE(InfoExtractor):
             'uploader': 'takuya0301',
             'uploader_id': '2698420',
             'upload_date': '20131123',
-            'timestamp': 1385182762,
+            'timestamp': int, # timestamp is unstable
```
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: >- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

Comment (pawitp) [reply youtube_dl/extractor/niconico.py]: Sorry, forgot to ran it again after the changes.

**Pull request merged**